### PR TITLE
Ember: Fix broken import

### LINKED ISSUE #20653: [Bug]: Fresh install Storybook with Ember framework produces a ModuleNotFoundError
### Describe the bug

Hello 👋 

First of all, thank you for all the great work you do for the community 🙏 

Today, I tried a fresh install of storybook for ember framework using the following guide : https://storybook.js.org/docs/ember/get-started/install 

At the step `npm run storybook` it produces this error : 
```
ModuleNotFoundError: Module not found: Error: Can't resolve '@ember/component' in '/Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/dist/esm/client/preview'
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/webpack/lib/Compilation.js:2016:28
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/webpack/lib/NormalModuleFactory.js:798:13
    at eval (eval at create (/Users/myuser/Documents/workspace/ember-quickstart/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:10:1)
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/webpack/lib/NormalModuleFactory.js:270:22
    at eval (eval at create (/Users/myuser/Documents/workspace/ember-quickstart/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/webpack/lib/NormalModuleFactory.js:434:22
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/webpack/lib/NormalModuleFactory.js:116:11
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/webpack/lib/NormalModuleFactory.js:670:25
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/webpack/lib/NormalModuleFactory.js:855:8
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/webpack/lib/NormalModuleFactory.js:975:5
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/neo-async/async.js:6883:13
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/webpack/lib/NormalModuleFactory.js:958:45
    at finishWithoutResolve (/Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/Resolver.js:312:11)
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/Resolver.js:386:15
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/myuser/Documents/workspace/ember-quickstart/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/myuser/Documents/workspace/ember-quickstart/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:27:1)
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:87:43
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/myuser/Documents/workspace/ember-quickstart/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/forEachBail.js:16:12
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/AliasPlugin.js:125:14
    at next (/Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/forEachBail.js:14:3)
    at forEachBail (/Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/forEachBail.js:24:9)
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/AliasPlugin.js:51:5
    at _next0 (eval at create (/Users/myuser/Documents/workspace/ember-quickstart/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:8:1)
    at eval (eval at create (/Users/myuser/Documents/workspace/ember-quickstart/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:30:1)
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/myuser/Documents/workspace/ember-quickstart/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/myuser/Documents/workspace/ember-quickstart/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/myuser/Documents/workspace/ember-quickstart/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:27:1)
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:87:43
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/myuser/Documents/workspace/ember-quickstart/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/myuser/Documents/workspace/ember-quickstart/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at /Users/myuser/Documents/workspace/ember-quickstart/node_modules/enhanced-resolve/lib/DirectoryExistsPlugin.js:41:15
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
resolve '@ember/component' in '/Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/dist/esm/client/preview'
  Parsed request is a module
  using description file: /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/package.json (relative path: ./dist/esm/client/preview)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/dist/esm/client/preview/node_modules doesn't exist or is not a directory
      /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/dist/esm/client/node_modules doesn't exist or is not a directory
      /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/dist/esm/node_modules doesn't exist or is not a directory
      /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/dist/node_modules doesn't exist or is not a directory
      looking for modules in /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/node_modules
        single file module
          using description file: /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/package.json (relative path: ./node_modules/@ember/component)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/node_modules/@ember/component doesn't exist
            .mjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/node_modules/@ember/component.mjs doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/node_modules/@ember/component.js doesn't exist
            .jsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/node_modules/@ember/component.jsx doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/node_modules/@ember/component.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/node_modules/@ember/component.tsx doesn't exist
            .json
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/node_modules/@ember/component.json doesn't exist
            .cjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/node_modules/@ember/component.cjs doesn't exist
        /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/ember/node_modules/@ember/component doesn't exist
      /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@storybook/node_modules doesn't exist or is not a directory
      /Users/myuser/Documents/workspace/ember-quickstart/node_modules/node_modules doesn't exist or is not a directory
      looking for modules in /Users/myuser/Documents/workspace/ember-quickstart/node_modules
        single file module
          using description file: /Users/myuser/Documents/workspace/ember-quickstart/package.json (relative path: ./node_modules/@ember/component)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@ember/component doesn't exist
            .mjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@ember/component.mjs doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@ember/component.js doesn't exist
            .jsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@ember/component.jsx doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@ember/component.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@ember/component.tsx doesn't exist
            .json
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@ember/component.json doesn't exist
            .cjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@ember/component.cjs doesn't exist
        /Users/myuser/Documents/workspace/ember-quickstart/node_modules/@ember/component doesn't exist
      /Users/myuser/Documents/workspace/node_modules doesn't exist or is not a directory
      /Users/myuser/Documents/node_modules doesn't exist or is not a directory
      /Users/myuser/node_modules doesn't exist or is not a directory
      /Users/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
```

### To Reproduce

```shell
I can't create a "reproduction" using `npx sb@next sandbox` because my issue occurs during installation. Here is what I did:

- Use node 16 (which is still supported see [here](https://github.com/storybookjs/storybook/blob/next/code/frameworks/ember/package.json#L56))

nvm use 16
```
- Install latest version of Ember CLI (v4.9.2)
```
npm install -g ember-cli
```
- Create a fresh app Ember
```
ember new ember-quickstart --lang fr
```
- Check Ember app works fine
```
cd ember-quickstart
npm start
```
- Install storybook for Ember
```
npx storybook init
ember install @storybook/ember-cli-storybook
```
- Run storybook (it is when the error occurs)
```
npm run storybook
```
```


### System

```shell
Environment Info:

  System:
    OS: macOS 12.1
    CPU: (8) arm64 Apple M1
  Binaries:
    Node: 16.14.0 - ~/.nvm/versions/node/v16.14.0/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.13.2 - ~/.nvm/versions/node/v16.14.0/bin/npm
  Browsers:
    Chrome: 109.0.5414.87
    Firefox: 108.0.2
    Safari: 15.2
  npmPackages:
    @storybook/addon-actions: ^6.5.15 => 6.5.15
    @storybook/addon-essentials: ^6.5.15 => 6.5.15
    @storybook/addon-links: ^6.5.15 => 6.5.15
    @storybook/builder-webpack5: ^6.5.15 => 6.5.15
    @storybook/ember: ^6.5.15 => 6.5.15
    @storybook/ember-cli-storybook: ^0.6.0 => 0.6.0
    @storybook/manager-webpack5: ^6.5.15 => 6.5.15
```

### Additional context

I also tried to bootstrap an Ember Addon rather than an Ember App. It produces the same error.

### PR DESCRIPTION
**Pull request by yannbertrand**
Hopefully closes #20653 

## What I did

Ember [recommends to use `@glimmer/component` rather than `@ember/component`](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-classic-components.md). I switched the import and the package.json accordingly. I believe ember itself somehow import `@glimmer/component` so maybe the peer dependency is not needed. I couldn't launch the project myself locally so let's wait for CI check.

## How to test

- Create a fresh app Ember
```
npx --package ember-cli ember --version #ember-cli: 4.11.0
npx --package ember-cli ember new ember-quickstart
```
- Check Ember app works fine
```
cd ember-quickstart
npm start
```
- Install this storybook PR for Ember
```
npm install --save-dev yannbertrand/storybook#next
```
- Init Storybook
```
npx storybook --version #
npx storybook init
```
- Hopefully this command now works and render something
```
npm run storybook
```

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: code/frameworks/ember/src/client/preview/render.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { global } from '@storybook/global';
 import { dedent } from 'ts-dedent';
 import type { RenderContext } from '@storybook/types';
 // @ts-expect-error (Converted from ts-ignore)
```
Comment (dbendaou) [inline code/frameworks/ember/src/client/preview/render.ts:4]: I guess we can remove this line this types for `@glimmer` should be defined

Comment (yannbertrand) [reply code/frameworks/ember/src/client/preview/render.ts]: Done

**Comment by dbendaou** [comment]
Changes look good to me, did you had the opportunity to try with yarn link or something like this?

The CI was already not breaking last time and I've introduced the bug so I better be extra careful :)

**Comment by yannbertrand** [comment]
> Changes look good to me, did you had the opportunity to try with yarn link or something like this?
> 
> The CI was already not breaking last time and I've introduced the bug so I better be extra careful :)

Actually we tried, but I'm not sure on how to fully test this. Here are the steps I followed:

# Build and link `@storybook/ember`
```
cd storybook/scripts
yarn install 
cd ../code
yarn install
yarn build ember
cd frameworks/ember
npm link
```

# Init Ember project and Storybook
```
npx --package ember-cli ember new ember-quickstart
cd ember-quickstart

# I don't know how to link the CLI, so I `init` using production build
npx --package storybook@^6 storybook init # n, n

npm link @storybook/ember
npm run storybook
```
Looking into my `node_modules` folder, `@storybook/ember` seems properly linked.

But I get this error:
```
> ember-quickstart@0.0.0 storybook
> start-storybook -p 6006 -s dist

sh: start-storybook: command not found
```

How can I go further in my work?

**Comment by yannbertrand** [comment]
> But I get this error:
> 
> ```
> > ember-quickstart@0.0.0 storybook
> > start-storybook -p 6006 -s dist
> 
> sh: start-storybook: command not found
> ```

I believe this one happens because my local fork is Storybook@v7, so `start-storybook` doesn't exist anymore. Is there any way to fix Storybook@v6 before adressing v7?

**Comment by yannbertrand** [comment]
So I have no idea on how to fix this. The Ember example throws a lot of warnings. Upgrading to new versions totally breaks and we're not even in the latest major one.

If I understand it correctly, [Ember team looks to know about it](https://discord.com/channels/480462759797063690/662337973995700224/1047186587244564580). I hope they can fix it.

I'm closing my PR.

**Pull request closed without merging**